File: deepseek/3.Desarrollo Web Endpoint RESTful/python/endpoint.py
class Singleton:
    _instance = None

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super().__new__(cls)
        return cls._instance

# Prueba
s1 = Singleton()
s2 = Singleton()
print("¿s1 y s2 son la misma instancia?", s1 is s2)

File: deepseek/3.Desarrollo Web Endpoint RESTful/ruby/endpoint.rb
class Singleton
  @@instance = nil

  def self.instance
    @@instance ||= Singleton.new
  end

  private_class_method :new
end

# Prueba
s1 = Singleton.instance
s2 = Singleton.instance
puts "¿s1 y s2 son la misma instancia? #{s1 == s2}"